Purchases/Networking/HTTPClient.swift

```diff
@@ -224,6 +224,11 @@ private extension HTTPClient {
             }
         }
 
+        if DNSChecker.isBlockedAPIError(maybeNetworkError),
+            let resolvedHost = DNSChecker.blockedHostFromError(maybeNetworkError) {
+            Logger.error(NetworkStrings.blocked_network(newHost: resolvedHost))
```

Comment:
Not sure if I am not thinking this the same way, but this won't send this error to the end user, if we just log it, only the developer will see it. Right?

Reply:
Well maybe not to the end user, but at least send it to the developer as an error, so they can display troubleshooting to their users.

Reply:
I like that as a solution. The more easily a developer can surface the condition to an end user, the better.

Reply:
Just a couple more lines down:

```swift
        if let httpResponse = maybeHTTPResponse,
            let completionHandler = maybeCompletionHandler {
            let error = maybeJSONError ?? maybeNetworkError
            completionHandler(httpResponse.statusCode, httpResponse.jsonObject, error)
        }
```
We pass the `maybeNetworkError` to the caller 😄

Reply:
Sure sure, we complete with an error, but the error is still a `NSURLErrorCannotConnectToHost`, which can mean a lot of things, not only that we are being blocked. Because I am assuming it is possible to get that same error if, for example, the phone is connected to WIFI, but the WIFI doesn't have internet connection. I might be wrong.

What I mean is that if I am a developer, and I see an error with an underlying error that has a `Error Domain=kCFErrorDomainCFNetwork`, `Code=-1004` and a `NSErrorFailingURLKey`, I can't really show a message saying "Something is blocking RevenueCat's URLs", because I am not sure what caused the error, we are checking if it's blocked so we know the error is caused by the DNS, but we are not surfacing that to the developer apart from logging. If we create a custom error `RevenueCatBlockedError` that we create only when we check if the DNS is resolving to 0.0.0.0, they will be able to show the correct message to the user.

Apart from that, looking in the underlying error is very cumbersome in the hybrids. If we can also create the same error with the same error code in Android, responding to it will be easier in the hybrid SDKs.

Reply:
I updated this: https://github.com/RevenueCat/purchases-ios/pull/1113